ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -17,73 +17,84 @@
-	this.timeIntervalInSeconds = timeIntervalInSeconds;
-}
-
-public long getLastSuccessLogTime() {
-	return lastSuccessLogTime.longValue();
-}
-public void setLastSuccessLogTime(long lastSuccessLogTime) {
-	this.lastSuccessLogTime.set(lastSuccessLogTime);
-}
-public long getLastFailureLogTime() {
-	return lastFailureLogTime.longValue();
-}
-public void setLastFailureLogTime(long lastFailureLogTime) {
-	this.lastFailureLogTime.set(lastFailureLogTime);
-}
-
-public Logger getLogger() {
-	return logger;
-}
-public void setLogger(Logger logger) {
-	this.logger = logger;
-}
-public String[] getRoles() {
-	return roles;
-}
-public void setRoles(String... roles) {
-	this.roles = roles;
-}
-public int getBatchSize() {
-	return batchSize;
-}
-public void setBatchSize(int batchSize) {
-	this.batchSize = batchSize;
-}
-
-   public JobRunner() {
-   }
-
-   public String getJobId() {
-      return jobId;
-   }
-
-public void run(DatabaseClient client, InputStream csvRecords, InputStream jobControl) throws IOException {
-      DataMovementManager moveMgr = client.newDataMovementManager();
-      JSONDocumentManager jsonMgr = client.newJSONDocumentManager();
-      Long startTime = System.currentTimeMillis();
-      
+	private String jobId;
+	private String jobDirectory;
+	private int batchSize = 100;
+	private String[] roles;
+	private int timeIntervalInSeconds = 10;
+	private AtomicLong lastSuccessLogTime;
+	private AtomicLong lastFailureLogTime;
+	private ObjectNode status;
+
+	public ObjectNode getStatus() {
+		return status;
+	}
+
+	public void setStatus(ObjectNode status) {
+		this.status = status;
+	}
+
+	public int getTimeIntervalInSeconds() {
+		return timeIntervalInSeconds;
+	}
+
+	public void setTimeIntervalInSeconds(int timeIntervalInSeconds) {
+		this.timeIntervalInSeconds = timeIntervalInSeconds;
+	}
+
+	public long getLastSuccessLogTime() {
+		return lastSuccessLogTime.longValue();
+	}
+
+	public void setLastSuccessLogTime(long lastSuccessLogTime) {
+		this.lastSuccessLogTime.set(lastSuccessLogTime);
+	}
+
+	public long getLastFailureLogTime() {
+		return lastFailureLogTime.longValue();
+	}
+
+	public void setLastFailureLogTime(long lastFailureLogTime) {
+		this.lastFailureLogTime.set(lastFailureLogTime);
+	}
+
+	public String[] getRoles() {
+		return roles;
+	}
+
+	public void setRoles(String... roles) {
+		this.roles = roles;
+	}
+
+	public int getBatchSize() {
+		return batchSize;
+	}
+
+	public void setBatchSize(int batchSize) {
+		this.batchSize = batchSize;
+	}
+
+	public JobRunner() {
+	}
+
+	public String getJobId() {
+		return jobId;
+	}
+
+	public void run(DatabaseClient client, InputStream csvRecords, InputStream jobControl, JobLogger jobLogger) throws IOException {
+		DataMovementManager moveMgr = client.newDataMovementManager();
+		JSONDocumentManager jsonMgr = client.newJSONDocumentManager();
+		Long startTime = System.currentTimeMillis();
+		ObjectNode statusNode = new ObjectMapper().createObjectNode();
```

Comment:
Before setting the properties for each new call to status(), should the code reinitialize the statusNode with:
`statusNode = new ObjectMapper().createObjectNode();`
Otherwise, will properties from the previous call to status() that haven't been overwritten also appear into the next call?